pkg/client/cloudevents/grpcsource/pager.go

```diff
@@ -0,0 +1,60 @@
+package grpcsource
+
+import (
+	"context"
+
+	"github.com/openshift-online/maestro/pkg/api/openapi"
+	"k8s.io/klog/v2"
+)
+
+// MaxListPageSize is the maximum size of one page.
+// Default is 2000, this value is tested when the maestro server's memory limit is set to 1Gi and the
+// 2000 resources size is about 60Mi.
+// NOTE: This should be reset carefully, if a bigger value is used with the same test conditions, it will
+// lead to maestro server OOM.
+var MaxListPageSize int32 = 2000
```

Comment:
this is not configurable, right?

Reply:
should we decide this by https://github.com/kubernetes/apimachinery/blob/95b78024e3feada7739b40426690b4f287933fd8/pkg/apis/meta/internalversion/types.go#L63?

Reply:
it's configurable, we can use `grpcsource. MaxListPageSize` increase this value if it is needed